Tests/test_image.py

```diff
@@ -70,6 +70,10 @@ def test_invalid_image(self):
             im = io.BytesIO(b'')
         self.assertRaises(IOError, lambda: Image.open(im))
 
+    def test_bad_mode(self):
+        im = hopper()
+        self.assertRaises(ValueError, lambda: Image.open(im, "bad mode"))
```

Comment:
`im` there is `PIL.Image.Image` object. Does `Image.open(im)` make any sense?
I believe it should be something like:
```python
def test_bad_mode(self):
    self.assertRaises(ValueError, Image.open, "filename", "bad mode")
```

Reply:
Okay, I've updated the PR.